core/src/main/java/org/pentaho/di/connections/vfs/VFSDetailsComposite.java

```diff
@@ -33,4 +33,21 @@ public interface VFSDetailsComposite {
   default String validate() {
     return null;
   }
+
+  /**
+   * Gets the Object of SelectionAdapterFileDialogTextVar for the browse file dialog.
+   * @return Object of SelectionAdapterFileDialogTextVar or null otherwise.
+   * Defaults to {@code null}.
+   */
+  default Object getRootPathSelectionAdapter( Object textUiWidget ) {
+    return null;
+  }
+
+  /**
+   * Returns true if vfs connection can browse root path
+   * Defaults to {@code false}.
+   */
+  default boolean canBrowseRootPath() {
+    return false;
+  }
```

Comment:
This method is not needed, if assumed that `getRootPathSelectionAdapter` returns null when not to browse.

Reply:
Addressed